python/foglamp/services/core/api/service.py

```diff
@@ -398,11 +410,82 @@ def load_c_plugin(plugin: str, service_type: str) -> Dict:
         :Example:
             curl -X GET http://localhost:8081/foglamp/service/installed
     """
-    services = []
-    svc_prefix = 'foglamp.services.'
-    for root, dirs, files in os.walk(_FOGLAMP_ROOT + "/" + "services"):
-        for f in files:
-            if f.startswith(svc_prefix):
-                services.append(f.split(svc_prefix)[-1])
-
+    services = get_service_installed()
     return web.json_response({"services": services})
+
+
+async def update_service(request: web.Request) -> web.Response:
+    """ update service
+
+    :Example:
+        curl -sX PUT http://localhost:8081/foglamp/service/Notification/notification/update
+    """
+    _type = request.match_info.get('type', None)
+    name = request.match_info.get('name', None)
+    try:
+        _type = _type.lower()
+        if _type != 'notification':
+            raise ValueError("Invalid plugin type. Must be 'notification'")
+
+        # Check requested service name is installed or not
+        installed_services = get_service_installed()
+        if name not in installed_services:
+            raise KeyError("{} service is not yet installed. So update is not possible.".format(name))
+
+        storage_client = connect.get_storage_async()
+        # TODO: process_name ends with "_c" suffix
+        payload = PayloadBuilder().SELECT("id", "enabled", "schedule_name").WHERE(['process_name', '=', '{}_c'.format(
+            _type)]).payload()
+        result = await storage_client.query_tbl_with_payload('schedules', payload)
+        sch_info = result['rows']
+        sch_list = []
+        if sch_info and sch_info[0]['enabled'] == 't':
+            status, reason = await server.Server.scheduler.disable_schedule(uuid.UUID(sch_info[0]['id']))
+            if status:
+                _logger.warning("{} service is disabled as {} service is updating..".format(
+                    sch_info[0]['schedule_name'], name))
+                sch_list.append(sch_info[0]['id'])
+
+        # service update is running as a background task
+        loop = request.loop
+        request._type = _type
+        request._name = name
+        request._sch_list = sch_list
+        loop.call_later(1, do_update, request)
+    except KeyError as ex:
+        raise web.HTTPNotFound(reason=str(ex))
+    except ValueError as ex:
+        raise web.HTTPBadRequest(reason=str(ex))
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=str(ex))
+
+    return web.json_response({"message": "{} service update in process. Wait for few minutes to complete.".format(
+        name)})
+
+
+def do_update(request):
+    _logger.info("{} service update starts...".format(request._name))
+    name = "foglamp-service-{}".format(request._name.lower())
+    _platform = platform.platform()
+    stdout_file_path = common.create_log_file(action="update", plugin_name=name)
```

Comment:
if this accepts only 2 args, avoid key-worded args. especially as  `plugin_name` is there, this log action is for.